openlibrary/plugins/openlibrary/processors.py

```diff
@@ -109,6 +112,91 @@ def __call__(self, handler):
         return handler()
 
 
+class CacheablePathsProcessor:
+    """Processor for caching Infogami pages.
+
+    This processor determines if the current path matches any of the
+    patterns in the `paths_and_expiries` dictionary, and if so, it
+    attempts to return the page from cache.  If no cached page is found,
+    it will fetch the page using the usual handler.
+    """
+
+    # Maps paths of cacheable pages to the pages' expiry times
+    paths_and_expiries: dict[str, int] = {
+        '/collections': 10 * dateutil.MINUTE_SECS,
+        '/collections/([^/]+)': 5 * dateutil.MINUTE_SECS,
+    }
+
+    def __call__(self, handler):
+        def get_page() -> dict:
+            """Fetches the page using the usual handler.
+            In order to cache the page, it is returned as a dictionary.
+            """
+            _page = handler()
+            return dict(_page)
+
+        def get_cached_page(path: str) -> dict:
+            """Returns the page located at the given path.
+            Attempts to return a cached page, if one exists. The full
+            path is used as the cache key for the page.
+            """
+            mc = cache.memcache_memoize(
+                get_page,
+                path,
+                self.paths_and_expiries[match],
+                prethread=caching_prethread(),
+            )
+            _page = mc()
+
+            if not _page:
+                mc(_cache='delete')
+
+            return _page
+
```

Comment:
**Reviewer:** Please check the order of the conditionals that return `handler()`.

My goal is to order these such that we exit this method as quickly as possible if the page isn't being cached by this processor.